src/Take.Elephant.Tests/PartitionedSenderReceiverFacts .cs

```diff
@@ -0,0 +1,140 @@
+using AutoFixture;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Take.Elephant.Tests
+{
+    public abstract class PartitionedSenderReceiverFacts<TKey, TEvent> : FactsBase, IDisposable
+    {
+        private readonly CancellationTokenSource _cts;
+
+        public PartitionedSenderReceiverFacts()
+        {
+            _cts = new CancellationTokenSource(TimeSpan.FromSeconds(30));
+        }
+
+        public abstract (IEventStreamPublisher<string, Item>, IEventStreamConsumer<string, Item>) CreateStream();
+
+        public CancellationToken CancellationToken => _cts.Token;
+
+        protected virtual Item CreateItem()
+        {
+            return Fixture.Create<Item>();
+        }
+
+        [Fact(DisplayName = nameof(PublishNewItemSucceeds))]
+        public virtual async Task PublishNewItemSucceeds()
+        {
+            // Arrange
+            var (senderQueue, receiverQueue) = CreateStream();
```

Comment:
Hmmm, maybe in these new tests, you can change the variable names from sender and receiver queues to something related to kafka mechanism (like streamConsumer and streamConsumer). What do you think?

Reply:
I think it will be better. I will change here, thanks for the suggestion :)